Histórico de atualização

BidoOBido opened the pull request:
Adiciona Task semanal de histórico de atualização;
Ajusta Task diária para retornar também o nome da base;

BidoOBido committed:
Feature/tecnologia atual (#3)

* Renomeia run.py para main.py

* Adiciona exemplo para o arquivo de configuração

* Ajusta json

* Refatora classe de conexão com PostgreSQL

* Adiciona classes das tasks

* Adiciona classses dos notificadores

* Adiciona factory das tasks

* Adiciona factory das notificações

* Refatora classe main para utilizar factories

* Remove arquivos não utilizados

* Apenas notificar se tiver tarefas

* Adicionado path completo

* Adicionado hierarquia do json

* Alterado nome da classe

* Ajustado readme

* Ajuste arquivo json

* Adicionado ignore

* Adicionado comandos para docker

* Aumentado intervalo para 30 minutos

* Removido uses duplicado

* Ajustado SQL

* Feature/tecnologia atual (#1)

* Doc: Atualizado arquivo gitignore

* Doc: Adicionados modelos e refatorado para sua utilização

* Doc: Ajuste para gerar a url corretamente

* Doc: Ajusta metodo que valida intervalo de chamada

* Doc: Adicionado arquivo de tarefas em json

Adicionado também uso do pacote dotenv

* Doc: Passa a considerar hora de inicio e fim da execução

Co-authored-by: Mailon Bido <mailonbido@sysmo.com.br>

* Alterado configurações

* Adicionado comandos para criar imagem docker

* Adicionado dependência

* Atualizado documentação para utilização do docker

* Ajustado mensagem do stacktrace

* Implementa tipo de agendamento "Cron" (#4)

* Doc: Atualizado arquivo gitignore

* Doc: Adicionados modelos e refatorado para sua utilização

* Doc: Ajuste para gerar a url corretamente

* Doc: Ajusta metodo que valida intervalo de chamada

* Doc: Adicionado arquivo de tarefas em json

Adicionado também uso do pacote dotenv

* Doc: Passa a considerar hora de inicio e fim da execução

* Doc: Adicionada a capacidade de informar uma cron de execução

* Doc: Ajuste do arquivo requirements.txt

* Doc: Remove duplicação desnecessária de task

* Doc: Removidos imports não utilizados

* Doc: Ajuste do factory

Refatorado para que o factory de schedule seja realmente um factory

* Doc: Ajusta método is_time_to_run

Ajustes conforme sugerido em code review para aumentar a legibilidade

* Doc: Ajuste do factory

Ajustado nomenclatura do factory e refatorado para aumentar a legibilidade

* Doc: Implementadas multiplos agendamento

Passa a permitir cadastrar diversos agendamentos pra cada tarefa

Co-authored-by: Mailon Bido <mailonbido@sysmo.com.br>

* Feature/tecnologia 2 cron (#5)

* Doc: Adicionados modelos e refatorado para sua utilização

* Doc: Ajuste para gerar a url corretamente

* Doc: Ajusta metodo que valida intervalo de chamada

* Doc: Adicionado arquivo de tarefas em json

* Adicionado também uso do pacote dotenv

* Doc: Passa a considerar hora de inicio e fim da execução

* Doc: Adicionada a capacidade de informar uma cron de execução

* Doc: Ajuste do arquivo requirements.txt

* Doc: Remove duplicação desnecessária de task

* Doc: Removidos imports não utilizados

* Doc: Ajuste do factory

Refatorado para que o factory de schedule seja realmente um factory

* Doc: Ajusta método is_time_to_run

Ajustes conforme sugerido em code review para aumentar a legibilidade

* Doc: Ajuste do factory

Ajustado nomenclatura do factory e refatorado para aumentar a legibilidade

* Doc: Implementadas multiplos agendamento

Passa a permitir cadastrar diversos agendamentos pra cada tarefa

* Doc: Ajustado para escapar corretamente caracteres (#1)

Evitando assim erro na regex quando houverem caracteres especiais

Co-authored-by: Mailon Bido <mailonbido@sysmo.com.br>

Co-authored-by: Mailon Bido <mailonbido@sysmo.com.br>

* Doc: Ajustado encoding de leitura do arquivo de tasks

* Doc: Atualizado o arquivo env_example

* Doc: Adicionada nova task de notificação de clientes atualizados

* Feature/tecnologia atual (#6)

* Doc: Ajustado encoding de leitura do arquivo de tasks

* Doc: Atualizado o arquivo env_example

* Doc: Adicionada nova task de notificação de clientes atualizados

Co-authored-by: Mailon Bido <mailonbido@sysmo.com.br>

* Feature/date on title (#7)

* Doc: Ajustado encoding de leitura do arquivo de tasks

* Doc: Atualizado o arquivo env_example

* Doc: Adicionada nova task de notificação de clientes atualizados

* Doc: Adicionado suporte à concatenação de data atual no título da notificação

Co-authored-by: Mailon Bido <mailonbido@sysmo.com.br>

* Feature/date on title (#2)

* Doc: Adicionado suporte à concatenação de data atual no título da notificação

* Doc: Adiciona mais configurações para data no título

Co-authored-by: Mailon Bido <mailonbido@sysmo.com.br>

Co-authored-by: Rafael Gustavo Simon Werner <tecnologia3@sysmo.com.br>
Co-authored-by: Rafael Gustavo Simon Werner <rafaelgustavowerner@gmail.com>
Co-authored-by: Eduardo Uliana <tecnologia1@sysmo.com.br>
Co-authored-by: Mailon Bido <mailonbido@sysmo.com.br>

BidoOBido committed:
Merge branch 'eduardouliana:main' into main

BidoOBido committed:
Feature/tecnologia historico atualizacao (#4)

* Doc: Ajuste de coleta de atualizações

- Adicionado nome da base na coleta diária
- Adicionada coleta semanal com resumo

* Doc: Ajustada hora da task

Co-authored-by: Mailon Bido <mailonbido@sysmo.com.br>

RafaelWerner commented:
ué pq fechou? Seria interessante documentar o porque desistiu, se achou outra forma de fazer, ou se não vale a pena.. no futuro quando você olhar pra este PR ou ter uma ideia parecida talvez você não lembre do motivo.

BidoOBido committed:
Feature/tecnologia historico atualizacao (#5)

* Doc: Ajuste de coleta de atualizações

- Adicionado nome da base na coleta diária
- Adicionada coleta semanal com resumo

* Doc: Ajustada hora da task

* Doc: Removida localização de data

- Ao rodar no docker/linux houve problema com a localização, dado que não é uma função crítica a mesma foi removida no momento
- Quando houver a necessidade é preciso investigar a razão do erro

Co-authored-by: Mailon Bido <mailonbido@sysmo.com.br>

BidoOBido commented:
Reaberto após correção de erro com localização da data.

- Ao rodar no docker/linux houve problema com a localização, dado que não é uma função crítica a mesma foi removida no momento
- Quando houver a necessidade é preciso investigar a razão do erro

Commit:
Doc: Remoção de norificação redundante
